src/lib/asn1/asn1_time.cpp

```diff
@@ -230,14 +232,14 @@ std::string X509_Time::readable_string() const
    if(time_is_set() == false)
       throw Invalid_State("X509_Time::readable_string: No time set");
 
-   std::string output(24, 0);
-
-   std::sprintf(&output[0], "%04d/%02d/%02d %02d:%02d:%02d UTC",
-                year, month, day, hour, minute, second);
-
-   output.resize(23); // remove trailing null
-
-   return output;
+   // desired format: "%04d/%02d/%02d %02d:%02d:%02d UTC"
+   using namespace std;
+   stringstream out;
```

Comment:
Thanks for critical review. I decided to move `using namespace` to a sub-scope in order to clarify it's purpose.